src/pybind/mgr/status/module.py

```diff
@@ -279,17 +279,32 @@ def handle_osd_status(self, cmd):
             if bucket_filter and osd_id not in filter_osds:
                 continue
 
-            metadata = self.get_metadata('osd', "%s" % osd_id)
-            stats = osd_stats[osd_id]
-
-            osd_table.add_row([osd_id, metadata['hostname'],
-                               self.format_bytes(stats['kb_used'] * 1024, 5),
-                               self.format_bytes(stats['kb_avail'] * 1024, 5),
+            hostname = ""
+            kb_used = 0
+            kb_avail = 0
+
+            if osd_id in osd_stats:
+                metadata = self.get_metadata('osd', "%s" % osd_id)
+                stats = osd_stats[osd_id]
+                hostname = metadata['hostname']
+                kb_used = stats['kb_used'] * 1024
+                kb_avail = stats['kb_avail'] * 1024
+
+            state = ""
```

Comment:
This is the right format, but we don't need to write out a for loop like this: you can do it in one line with `",".join(osd['state'])`

Reply:
done.